CHANGELOG.md

```diff
@@ -2,6 +2,26 @@
 
 ## next
 
+* First pass at describing (and doc-generating) API global information
+  * Inside a `Praxis::ApiDefinition.define` block one can now specify a few things about the API by using:
+    * info("1.0") `<block>` - Which will apply to a particular version only
+    * info `<block>` - Which will be inherited by any existing API version
+    * The current pieces of information that can be defined in the block are: `name`, `title`, `description` and `basepath`. See [this](https://github.com/rightscale/praxis/blob/master/spec/spec_app/design/api.rb) for details
+  * NOTE: This information is output to the JSON files, BUT not used in the doc browser yet.
+* Changed the doc generation and browser to use "ids" instead of "names" for routes and generated files.
+  * Currently, "ids" are generated using dashes instead of double colons (instead of random ids). This closes issue #31.
+* Added the definition and handling of canonical urls for API resources
+  * One can now specify which action URL to be considered as the canonical resource href:
+    * by using `canonical_path_action <action_name>` at the top of the resource definition class
+    * or by using `canonical_path` within the action definition itself.
+    * See the [instances](https://github.com/rightscale/praxis/blob/master/spec/spec_app/design/resources/instances.rb) resource definition for an example.
+  * With a canonical href defined, one can then both generate and parse them by using:
+    * `.to_href(<named arguments hash>)  =>  <href String>`
+    * `.parse_href( <href String> )  => < named arguments hash >`. Note: The returned arguments are properly typed-coerced.
+    * These helpers can be accessed from:
+      * the `definition` object in the controller instance (i.e., `self.definition.to_href(id: 1). )
+      * or through the class-level methods in the resource definition (i.e. `MyApiResource.parse_href("/my_resource/1")` )
+
```

Comment:
hmm...I'm not sure there's always a 1:1 from a MediaType to a resource ... although it is really the common case. Maybe we need to mull this over more. Might be good to bring that up in the current  email thread of the dist list about canonical urls...